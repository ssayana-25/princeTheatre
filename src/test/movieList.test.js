import React from 'react';
import MovieList from '../components/movieList.jsx';
import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

afterEach(cleanup);

//It can be validated from the snapshot that the prices are compared and the user is show the website with the lowest price.
it("Movie data is being populated and prices compared", ()=>{
    const tree= renderer.create(<MovieList poster="" title="Bahubali"
    website1Price="31" website2Price="50" website1="Cinema World" website2="Film World"></MovieList>).toJSON();
    expect(tree).toMatchSnapshot();
})

