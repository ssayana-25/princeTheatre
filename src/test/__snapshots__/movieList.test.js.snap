// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie data is being populated and prices compared 1`] = `
<div
  className="card p-3"
>
  <img
    alt="Bahubali"
    src=""
  />
  <div
    className="card-body"
  >
    <div
      className="pt-2 pb-1 movie-title font-weight-bold"
    >
      Bahubali
    </div>
    <div
      className="fwPrice row"
    >
      <div
        className="col-6"
      >
        Film World
        :
      </div>
      <div
        className="col-6"
      >
        50
      </div>
    </div>
    <div
      className="cwPrice row"
    >
      <span
        className="col-6"
      >
        Cinema World
        :
      </span>
      <span
        className="col-6"
      >
        31 low!
      </span>
    </div>
  </div>
</div>
`;
